src/main/java/com/google/api/codegen/metadatagen/PackageMetadataGeneratorTool.java

```diff
@@ -0,0 +1,173 @@
+import com.google.api.tools.framework.tools.ToolOptions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+/** Tool to generate package metadata. */
+public class PackageMetadataGeneratorTool {
+
+  private static final Map<String, List<String>> SNIPPETS =
+      new ImmutableMap.Builder<String, List<String>>()
+          .put(
+              "python",
+              Lists.newArrayList(
+                  "py/setup.py.snip",
+                  "py/README.rst.snip",
+                  "py/PUBLISHING.rst.snip",
+                  "py/MANIFEST.in.snip"))
+          .build();
+
+  private static final Map<String, PackageCopier> COPIERS =
+      new ImmutableMap.Builder<String, PackageCopier>()
+          .put("python", new PythonPackageCopier())
+          .build();
+
+  public static void main(String[] args) throws Exception {
+    Options options = new Options();
+    options.addOption("h", "help", false, "show usage");
+    options.addOption(
+        Option.builder()
+            .longOpt("descriptor_set")
+            .desc("The descriptor set representing the compiled input protos.")
+            .hasArg()
+            .argName("DESCRIPTOR-SET")
+            .required(true)
+            .build());
+    options.addOption(
+        Option.builder()
+            .longOpt("service_yaml")
+            .desc("The service YAML configuration file or files.")
+            .hasArg()
+            .argName("SERVICE-YAML")
+            .required(true)
+            .build());
+    options.addOption(
+        Option.builder("i")
+            .longOpt("input")
+            .desc("The input directory containing the gRPC package.")
+            .hasArg()
+            .argName("INPUT-DIR")
+            .required(true)
+            .build());
+    options.addOption(
+        Option.builder("o")
+            .longOpt("output")
+            .desc("The directory in which to output the generated config.")
+            .hasArg()
+            .argName("OUTPUT-DIR")
+            .required(true)
+            .build());
+    options.addOption(
+        Option.builder("d")
+            .longOpt("dependencies_config")
+            .desc("The dependencies configuration file.")
+            .hasArg()
+            .argName("DEPENDENCIES-FILE")
+            .required(true)
+            .build());
+    options.addOption(
+        Option.builder("d")
+            .longOpt("defaults_config")
+            .desc("The defaults configuration file.")
+            .hasArg()
+            .argName("DEFAULTS-FILE")
+            .required(true)
+            .build());
+    options.addOption(
+        Option.builder()
+            .longOpt("python_pkg_config")
```

Comment:
Why it's Python specific?  Is there anything special on Python?

I'm a bit concerned about this because it's unclear what else parameters will be added when supporting more languages. Adding more language specific parameters would lead to a combination problem.